test/test_torch.py

```diff
@@ -12507,71 +12503,55 @@ def test_put_empty(self, device):
 
-    @onlyCPU
     def test_scatter_reduce_operations_to_large_input(self, device):
         index = torch.tensor([[1], [2]], device=device, dtype=torch.long)
-        test_data = [
-            (torch.zeros(4, 4, device=device, dtype=torch.float32),
-             torch.ones(2, 2, device=device, dtype=torch.float32),
-             torch.tensor([[0, 0, 0, 0],
-                           [1, 0, 0, 0],
-                           [1, 0, 0, 0],
-                           [0, 0, 0, 0]],
-                          device=device, dtype=torch.float32), "add"),
-            (torch.zeros(4, 4, device=device, dtype=torch.float32),
-             torch.ones(2, 2, device=device, dtype=torch.float32),
-             torch.tensor([[0, 0, 0, 0],
-                           [-1, 0, 0, 0],
-                           [-1, 0, 0, 0],
-                           [0, 0, 0, 0]], device=device, dtype=torch.float32), "subtract"),
-            (torch.tensor([2], device=device, dtype=torch.float32).repeat(4, 4),
-             torch.tensor([2], device=device, dtype=torch.float32).repeat(2, 2),
-             torch.tensor([[2, 2, 2, 2],
-                           [4, 2, 2, 2],
-                           [4, 2, 2, 2],
-                           [2, 2, 2, 2]], device=device, dtype=torch.float32), "multiply"),
-            (torch.tensor([2], device=device, dtype=torch.float32).repeat(4, 4),
-             torch.tensor([2], device=device, dtype=torch.float32).repeat(2, 2),
-             torch.tensor([[2, 2, 2, 2],
-                           [1, 2, 2, 2],
-                           [1, 2, 2, 2],
-                           [2, 2, 2, 2]], device=device, dtype=torch.float32), "divide")
-        ]
+        for dtype in torch.testing.get_all_fp_dtypes(
+                include_bfloat16=device.startswith('cuda')) + torch.testing.get_all_complex_dtypes():
+            test_data = [
+                (torch.zeros(4, 4, device=device, dtype=dtype),
+                 torch.ones(2, 2, device=device, dtype=dtype),
+                 torch.tensor([[0, 0, 0, 0],
+                               [1, 0, 0, 0],
+                               [1, 0, 0, 0],
+                               [0, 0, 0, 0]],
+                              device=device, dtype=dtype), "add"),
+                (torch.tensor([2], device=device, dtype=dtype).repeat(4, 4),
+                 torch.tensor([6], device=device, dtype=dtype).repeat(2, 2),
+                 torch.tensor([[2, 2, 2, 2],
+                               [12, 2, 2, 2],
+                               [12, 2, 2, 2],
+                               [2, 2, 2, 2]], device=device, dtype=dtype), "multiply"),
+            ]
 
-        for input, src, result, operation in test_data:
-            input.scatter_(0, index, src, reduce=operation)
-            self.assertEqual(input, result)
+            for input, src, result, operation in test_data:
+                if operation == "multiply" and torch.is_complex(input):
+                    continue
```

Comment:
can you please check do `self.assertRaisesRegexp` here and in similar places?

Reply:
I'm doing this in a separate test here: https://github.com/pytorch/pytorch/pull/41977/files#diff-9996665f82f52030836eb8657057cfadR12591